images/linstor-scheduler-extender/patches/stork/stork.patch

```diff
@@ -0,0 +1,135 @@
+diff --git a/pkg/extender/extender.go b/pkg/extender/extender.go
+index 73c8c307a..3224dd2dc 100644
+--- a/pkg/extender/extender.go
++++ b/pkg/extender/extender.go
+@@ -3,8 +3,11 @@ package extender
+ import (
+ 	"context"
+ 	"encoding/json"
++	"errors"
+ 	"fmt"
+ 	"net/http"
++	"net/http/httputil"
++	"os"
+ 	"strconv"
+ 	"strings"
+ 	"sync"
+@@ -77,6 +80,8 @@ var (
+ type Extender struct {
+ 	Recorder record.EventRecorder
+ 	Driver   volume.Driver
++	CertFile string `json:"cert-file"`
++	KeyFile  string `json:"key-file"`
+ 	server   *http.Server
+ 	lock     sync.Mutex
+ 	started  bool
+@@ -94,7 +99,30 @@ func (e *Extender) Start() error {
+ 	e.server = &http.Server{Addr: ":8099"}
+ 	http.HandleFunc("/", e.serveHTTP)
+ 	go func() {
+-		if err := e.server.ListenAndServe(); err != http.ErrServerClosed {
++		err := error(nil)
++		tlsMode := len(e.CertFile) > 0 && len(e.KeyFile) > 0
++
++		if tlsMode {
++			if _, exist := os.Stat(e.CertFile); errors.Is(exist, os.ErrNotExist) {
++				log.Warnf("CertFile `%s` does not exists. Launch as HTTP", e.CertFile)
++				tlsMode = false
++			}
++
++			if _, exist := os.Stat(e.KeyFile); errors.Is(exist, os.ErrNotExist) {
++				log.Warnf("KeyFile `%s` does not exists. Launch as HTTP", e.KeyFile)
++				tlsMode = false
++			}
++		}
++
++		if tlsMode {
++			log.Warn("Starting as HTTPS on :8099")
++			err = e.server.ListenAndServeTLS(e.CertFile, e.KeyFile)
++		} else {
++			log.Warn("Starting as HTTP on :8099")
++			err = e.server.ListenAndServe()
++		}
++
++		if err != http.ErrServerClosed {
+ 			log.Panicf("Error starting extender server: %v", err)
+ 		}
+ 	}()
+@@ -149,6 +177,14 @@ func (e *Extender) getHostname(node *v1.Node) string {
+ }
+
+ func (e *Extender) processFilterRequest(w http.ResponseWriter, req *http.Request) {
++	payload, err := httputil.DumpRequest(req, true)
++	if err != nil {
++		log.Errorf("Error debugging filter request: %v", err)
++		http.Error(w, err.Error(), http.StatusInternalServerError)
++		return
++	}
++	log.Tracef("[filter] received request: %s", string(payload))
++
+ 	decoder := json.NewDecoder(req.Body)
+ 	defer func() {
+ 		if err := req.Body.Close(); err != nil {
+@@ -164,6 +200,13 @@ func (e *Extender) processFilterRequest(w http.ResponseWriter, req *http.Request
+ 		return
+ 	}
+
++	args, err = fillNodesByNames(args)
++	if err != nil {
++		log.Errorf("Cannot fill nodes: %+v", err)
```

Comment:
Here, and in other places, `%v` is good enough format in order to use output of `err.Error()`. Getting that string by yourself and rendering it with `%s` is even better.